### PR TITLE
[SYCL] Use SG broadcast to assert uniformity in `uniform<T>`'s ctor

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

**Comment by sarnex** [comment]
Will approve once E2E tests pass

**Comment by sarnex** [comment]
@aelovikov-intel Looks like a header include problem, all invokesimd tests fail:

```

# | /__w/llvm/llvm/toolchain/bin/../include/sycl/ext/oneapi/experimental/uniform.hpp:85:13: error: use of undeclared identifier 'group_broadcast'
# |    85 |       : Val(group_broadcast(sycl::ext::oneapi::this_work_item::get_sub_group(),
```

File: sycl/include/sycl/ext/oneapi/experimental/uniform.hpp [hunk 1]
```diff
@@ -67,11 +69,22 @@ template <class T> class uniform {
...
+  //
+  // Use sub-group broadcast to communicate the uniformity property to the
+  // compiler. Note that this is only capable of propagating the information
+  // "down" to the users in the def-use chain. E.g.
+  //
+  //   auto val = load(divergent-ptr) // Cannot be changed to a scalar load.
+  //   auto u = uniform(val)
+  //   use(u);                        // Can be optimized based on uniformity.
+  explicit uniform(T x) noexcept
+      : Val(group_broadcast(sycl::ext::oneapi::this_work_item::get_sub_group(),
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/ext/oneapi/experimental/uniform.hpp:86]: `group_broadcast` can throw an exception but ctor is `noexcept`

**Review by sarnex — APPROVED** [review]
EDIT: see below

**Pull request closed without merging**